Pnpm Audit Analyzer registry url

**Is your feature request related to a problem? Please describe.**
Pnpm Audit Analyzer throws `ECONNREFUSED` error due to npm registry.
It would be great to be able to specify my own npm package registry url.

**Describe the solution you'd like**
Suggesting to add a new optional Command Line Argument called `--npmRegistry <url>` or just `--registry <url>`.
Default value can be: https://registry.npmjs.org/

**Additional context**
![Cursor_and_https___jenkins_dhl_com_job_eCS_CONNECT_job_Connect-Web-UI_job_CI_job_master_57_artifact_owasp_log](https://user-images.githubusercontent.com/18481498/231713556-06507fd5-6d98-49cb-b9ac-7c723f133ab1.png)

Hi !
I have the same issue.
First, I didn't understand why the registry had to be https://registry.npmjs.org/ ( https://github.com/jeremylong/DependencyCheck/blob/33c4c98e868b6b34d32ada870de79f9fa56aa42e/core/src/main/java/org/owasp/dependencycheck/analyzer/PnpmAuditAnalyzer.java#L207C31-L207C31 ). I modified my version of PNPM so that it ignores the `--registry` argument. The request worked correctly but the response is not suitable. I have this exception:

> org.owasp.dependencycheck.analyzer.exception.AnalysisException: Failed to parse C:\jenkins\workspace\blabla\blabla\pnpm-lock.yaml file from the NPM Audit API (PnpmAuditAnalyzer) .
     at org.owasp.dependencycheck.analyzer.PnpmAuditAnalyzer.analyzePackage (PnpmAuditAnalyzer.java:271)
     at org.owasp.dependencycheck.analyzer.PnpmAuditAnalyzer.analyzeDependency (PnpmAuditAnalyzer.java:93)
     at org.owasp.dependencycheck.analyzer.AbstractAnalyzer.analyze (AbstractAnalyzer.java:131)
     at org.owasp.dependencycheck.AnalysisTask.call (AnalysisTask.java:88)
     at org.owasp.dependencycheck.AnalysisTask.call (AnalysisTask.java:37)
     at java.util.concurrent.FutureTask.run (FutureTask.java:264)
     at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
     at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
     at java.lang.Thread.run (Thread.java:840)
Caused by: org.json.JSONException: No value for github_advisory_id

The company I work at uses JFrog Artifactory and the `pnpm audit` command works perfectly on my computer.
For the moment I have no other solution than to allow connection to the default registry.

--- a/core/src/main/java/org/owasp/dependencycheck/analyzer/PnpmAuditAnalyzer.java
+++ b/core/src/main/java/org/owasp/dependencycheck/analyzer/PnpmAuditAnalyzer.java
@@ -61,6 +61,11 @@ public class PnpmAuditAnalyzer extends AbstractNpmAnalyzer {
     public static final String PNPM_PACKAGE_LOCK = "pnpm-lock.yaml";
 
     /**
+     * The default registry parameter to pass to the pnpm audit execution.
+     */
+    public static final String DEFAULT_REGISTRY = "https://registry.npmjs.org/";
+
+    /**
      * Filter that detects files named "pnpm-lock.yaml"
      */
     private static final FileFilter LOCK_FILE_FILTER = FileFilterBuilder.newInstance()
@@ -202,9 +207,9 @@ public class PnpmAuditAnalyzer extends AbstractNpmAnalyzer {
             }
             // pnpm audit returns a json compliant with NpmAuditParser
             args.add("--json");
-            // ensure we are using the right registry despite .npmrc
+            // ensure we are using the right registry despite .npmrc, but allow override
             args.add("--registry");
-            args.add("https://registry.npmjs.org/");
+            args.add(getSettings().getString(Settings.KEYS.ANALYZER_PNPM_AUDIT_REGISTRY, DEFAULT_REGISTRY));
             final ProcessBuilder builder = new ProcessBuilder(args);
             builder.directory(folder);
             // Workaround 64k limitation of InputStream, redirect stdout to a file that we will read later

--- a/core/src/main/resources/dependencycheck.properties
+++ b/core/src/main/resources/dependencycheck.properties
@@ -118,6 +118,7 @@ analyzer.node.package.enabled=true
 analyzer.node.audit.enabled=true
 analyzer.yarn.audit.enabled=true
 analyzer.pnpm.audit.enabled=true
+analyzer.pnpm.audit.registry=https://registry.npmjs.org/
 analyzer.golang.dep.enabled=true
 analyzer.retirejs.enabled=true
 analyzer.retirejs.repo.validforhours=24

--- a/utils/src/main/java/org/owasp/dependencycheck/utils/Settings.java
+++ b/utils/src/main/java/org/owasp/dependencycheck/utils/Settings.java
@@ -429,6 +429,10 @@ public final class Settings {
          */
         public static final String ANALYZER_PNPM_AUDIT_ENABLED = "analyzer.pnpm.audit.enabled";
         /**
+         * The properties key for the Pnpm registry url.
+         */
+        public static final String ANALYZER_PNPM_AUDIT_REGISTRY = "analyzer.pnpm.audit.registry";
+        /**
          * The properties key for supplying the URL to the Node Audit API.
          */
         public static final String ANALYZER_NODE_AUDIT_URL = "analyzer.node.audit.url";

